mscore/file.cpp

```diff
@@ -252,7 +252,7 @@ static bool readScoreError(const QString& name, Score::FileError error, bool ask
 
 bool MuseScore::checkDirty(MasterScore* s)
       {
-      if (s->dirty() || s->created()) {
+      if (s->dirty() || (s->created() && !s->startedEmpty())) {
```

Comment:
Instead of checking this new variable, isn't better to check the size of the undo stack? If that stack is empty nothing happened so it is save to disable the "Save" command.

Reply:
What if the user made changes and undoed them?

Reply:
> Instead of checking this new variable, isn't better to check the size of the undo stack?

What you're suggesting is actually the way it used to work, but it was a problem because MuseScore would fail to prompt the user to save a new score containing user-entered metadata (title, composer, key signature, tempo, etc.) if it had not yet been modified after being created (i.e., if it had an empty undo stack).

That bug was fixed in [#301929](https://musescore.org/en/node/301929), resulting in the side effect that's being addressed in this PR.

Reply:
> What if the user made changes and undoed them?

Then the user is not prompted to save the score. This behavior is consistent with all other cases where the user has made changes and then undone them. For example, if the user opens an existing score, makes an edit, undoes it, and then closes the score, he is not prompted to save changes (and rightly so).

Reply:
I understand, thanks!